File: tests/algorithms/x/Java/maths/least_common_multiple.java
public class Main {

    static long gcd(long a, long b) {
        long x = a >= 0 ? a : -a;
        long y_1 = b >= 0 ? b : -b;
        while (y_1 != 0) {
            long temp_1 = Math.floorMod(x, y_1);
            x = y_1;
            y_1 = temp_1;
        }
        return x;
    }

    static long lcm_slow(long a, long b) {
        long max = a >= b ? a : b;
        long multiple_1 = max;
        while ((Math.floorMod(multiple_1, a) != 0) || (Math.floorMod(multiple_1, b) != 0)) {
            multiple_1 = multiple_1 + max;
        }
        return multiple_1;
    }

    static long lcm_fast(long a, long b) {
        return ((Number)((Math.floorDiv(a, gcd(a, b))))).intValue() * b;
    }
    public static void main(String[] args) {
        {
            long _benchStart = _now();
            long _benchMem = _mem();
            System.out.println(_p(lcm_slow(5, 2)));
            System.out.println(_p(lcm_slow(12, 76)));
            System.out.println(_p(lcm_fast(5, 2)));
            System.out.println(_p(lcm_fast(12, 76)));
            long _benchDuration = _now() - _benchStart;
            long _benchMemory = _mem() - _benchMem;
            System.out.println("{");
            System.out.println("  \"duration_us\": " + _benchDuration + ",");
            System.out.println("  \"memory_bytes\": " + _benchMemory + ",");
            System.out.println("  \"name\": \"main\"");
            System.out.println("}");
            return;
        }
    }

    static boolean _nowSeeded = false;
    static int _nowSeed;
    static int _now() {
        if (!_nowSeeded) {
            String s = System.getenv("MOCHI_NOW_SEED");
            if (s != null && !s.isEmpty()) {
                try { _nowSeed = Integer.parseInt(s); _nowSeeded = true; } catch (Exception e) {}
            }
        }
        if (_nowSeeded) {
            _nowSeed = (int)((_nowSeed * 1664525L + 1013904223) % 2147483647);
            return _nowSeed;
        }
        return (int)(System.nanoTime() / 1000);
    }

    static long _mem() {
        Runtime rt = Runtime.getRuntime();
        rt.gc();
        return rt.totalMemory() - rt.freeMemory();
    }

    static String _p(Object v) {
        if (v == null) return "<nil>";
        if (v.getClass().isArray()) {
            if (v instanceof int[]) return java.util.Arrays.toString((int[]) v);
            if (v instanceof long[]) return java.util.Arrays.toString((long[]) v);
            if (v instanceof double[]) return java.util.Arrays.toString((double[]) v);
            if (v instanceof boolean[]) return java.util.Arrays.toString((boolean[]) v);
            if (v instanceof byte[]) return java.util.Arrays.toString((byte[]) v);
            if (v instanceof char[]) return java.util.Arrays.toString((char[]) v);
            if (v instanceof short[]) return java.util.Arrays.toString((short[]) v);
            if (v instanceof float[]) return java.util.Arrays.toString((float[]) v);
            return java.util.Arrays.deepToString((Object[]) v);
        }
        if (v instanceof Double || v instanceof Float) {
            double d = ((Number) v).doubleValue();
            if (d == Math.rint(d)) return String.valueOf((long) d);
            return String.valueOf(d);
        }
        return String.valueOf(v);
    }
}

File: tests/algorithms/x/Kotlin/maths/least_common_multiple.kt
import java.math.BigInteger

fun _numToStr(v: Number): String {
    val d = v.toDouble()
    val i = d.toLong()
    return if (d == i.toDouble()) i.toString() else d.toString()
}

var _nowSeed = 0L
var _nowSeeded = false
fun _now(): Long {
    if (!_nowSeeded) {
        System.getenv("MOCHI_NOW_SEED")?.toLongOrNull()?.let {
            _nowSeed = it
            _nowSeeded = true
        }
    }
    return if (_nowSeeded) {
        _nowSeed = (_nowSeed * 1664525 + 1013904223) % 2147483647
        kotlin.math.abs(_nowSeed)
    } else {
        kotlin.math.abs(System.nanoTime())
    }
}

fun toJson(v: Any?): String = when (v) {
    null -> "null"
    is String -> "\"" + v.replace("\"", "\\\"") + "\""
    is Boolean, is Number -> v.toString()
    is Map<*, *> -> v.entries.joinToString(prefix = "{", postfix = "}") { toJson(it.key.toString()) + ":" + toJson(it.value) }
    is Iterable<*> -> v.joinToString(prefix = "[", postfix = "]") { toJson(it) }
    else -> toJson(v.toString())
}

fun gcd(a: Int, b: Int): Int {
    var x: Int = (if (a >= 0) a else 0 - a.toInt()).toInt()
    var y: Int = (if (b >= 0) b else 0 - b.toInt()).toInt()
    while (y != 0) {
        var temp: Int = (Math.floorMod(x, y)).toInt()
        x = y
        y = temp
    }
    return x
}

fun lcm_slow(a: Int, b: Int): Int {
    var max: Int = (if (a >= b) a else b).toInt()
    var multiple: Int = (max).toInt()
    while (((Math.floorMod(multiple, a)) != 0) || ((Math.floorMod(multiple, b)) != 0)) {
        multiple = multiple + max
    }
    return multiple
}

fun lcm_fast(a: Int, b: Int): Int {
    return (a / gcd(a, b)) * b
}

fun main() {
    run {
        System.gc()
        val _startMem = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()
        val _start = _now()
        println(_numToStr(lcm_slow(5, 2)))
        println(_numToStr(lcm_slow(12, 76)))
        println(_numToStr(lcm_fast(5, 2)))
        println(_numToStr(lcm_fast(12, 76)))
        System.gc()
        val _end = _now()
        val _endMem = Runtime.getRuntime().totalMemory() - Runtime.getRuntime().freeMemory()
        val _durationUs = (_end - _start) / 1000
        val _memDiff = kotlin.math.abs(_endMem - _startMem)
        val _res = mapOf("duration_us" to _durationUs, "memory_bytes" to _memDiff, "name" to "main")
        println(toJson(_res))
    }
}
